models/src/Finder.cpp

```diff
@@ -0,0 +1,125 @@
+#include "../include/Finder.hpp"
+#include <RNG.hpp>
+const int NUM_MOVEMENT_STATES = 4;
+const float AGGROTIME = 4;
+const float DEFAULT_SPEED = 2.5;
+const float AUTOSWITCH = 1;
+const bool DEFAULT_WALLPASS = true;
+const int SIGHT_RANGE = 8;
+const sf::Vector2f DEFAULT_START(9.5, 9.5);
+
+Finder::Finder()		
+{
+	_position = (DEFAULT_START);
+	_enemySpeed = (DEFAULT_SPEED);
+	moveState = (EnemyMoveState::north);
+	_switchTime = (AUTOSWITCH);
+	type = (EnemyType::EFinder);
+	_wallPass = (DEFAULT_WALLPASS);
+	changeMoveState();
+}
+Finder::Finder(sf::Vector2f start)		
+{
+	_position = (start);
+	_enemySpeed = (DEFAULT_SPEED);
+	moveState = (EnemyMoveState::north);
+	_switchTime = (AUTOSWITCH);
+	type = (EnemyType::EFinder);
+	_wallPass = (DEFAULT_WALLPASS);
+	changeMoveState();
+}
+
+Finder::~Finder(){
+	return;
+}
+
+void Finder::changeMoveState()
+{
+	this->moveState =(EnemyMoveState)(RNG::getRandomNumber(0,NUM_MOVEMENT_STATES-1));
+}
+void Finder::changeMoveState(const Map &map, const Player &player)
+{
+	this->moveState = findpath(map, sf::Vector2i(this->position()), sf::Vector2i(player.position()));
+}
+
+void Finder::update(float deltaTime, const Map &map, const Player &player)
+{
+	_switchTime -= deltaTime;
+	double mf;
+	double xdec, ydec;
+	xdec = modf(this->position().x, &mf);
+	ydec = modf(this->position().y, &mf);
+	sf::Vector2i dist = distance(sf::Vector2i(this->position()), sf::Vector2i(player.position()));
+	if(abs(dist.y) < SIGHT_RANGE && abs(dist.x) < SIGHT_RANGE){
+		if((xdec <0.55 && xdec > 0.4) && (ydec <0.55 && ydec > 0.4) &&  ((int)(this->position().x)%2 || (int)(this->position().y)%2))
+			this->changeMoveState(map, player);
+	}else{
+		if(_switchTime <= 0)
+		{	
+			this->changeMoveState();
+			_switchTime = (RNG::getRandomNumber(0, (int)(AUTOSWITCH))) +1 ;
+		}
+	}
+
+	this->move(deltaTime, map);
+}
+
+void Finder::changeAggression()
+{
+	if(_aggression == 3 && type == EnemyType::EBallom)
+	{
+		_enemySpeed = DEFAULT_SPEED + 1.5;
+		type = EnemyType::EAggroBallom;
+	}
+	else
+	{
+		_enemySpeed = DEFAULT_SPEED;
+		type = EnemyType::EBallom;
+	}
+		_aggression = AGGROTIME;		
+}
+sf::Vector2i Finder::distance(sf::Vector2i start, sf::Vector2i end){
```

Comment:
Just format the rest of this file correctly. There are a myriad of problems that you should be able to spot yourself. Especially mind the indentation.